### ISSUE TITLE
Datasource explore link does not work when ROOT_URL / SERVE_FROM_SUB_PATH is used

### ISSUE THREAD

**Issue by Mario-Hofstaetter** [grafana/grafana#47443]
**What happened**:

Grafana is running behind nginx reverse proxy on sub-path `/grafana/`
When clicking the "Explore" button in prometheus datasource settings, the URL is incorrect, the subpath is missing.

![grafik](https://user-images.githubusercontent.com/33002073/162206130-ae34372f-7b7e-40c4-82e6-b5b96d650296.png)

**What you expected to happen**:

All links should honor `ROOT_URL` / `SERVE_FROM_SUB_PATH` settings.
This explore button seems rather new.

**How to reproduce it (as minimally and precisely as possible)**:

Run grafana using
```
      - "GF_SERVER_ROOT_URL=http://localhost/grafana"
      - "GF_SERVER_SERVE_FROM_SUB_PATH=true"
```

- Go to a (prometheus-) datasource settings page
- click "Explore" on the bottom of the page
-> Get `404`'d from reverse proxy (I don't know what happens without reverse proxy, probably grafana home page)

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.4.5
- Data source type & version: prometheus

**Comment by zuchka** [comment]
Thanks for creating this issue, @Mario-Hofstaetter 

I was not able to reproduce this using grafana 8.4.5 a fresh Ubuntu 20 vm. I would suggest you [try using the exact format listed in our NGINX tutorial](https://grafana.com/tutorials/run-grafana-behind-a-proxy/):

`root_url = %(protocol)s://%(domain)s:%(http_port)s/grafana/`

Here is my nginx config:

```
# this is required to proxy Grafana Live WebSocket connections.
map $http_upgrade $connection_upgrade {
  default upgrade;
  '' close;
}

server {
  listen 80;
  root /usr/share/nginx/www;
  index index.html index.htm;

  location /grafana/ {
   proxy_set_header Host $http_host; 
   proxy_pass http://localhost:3000/;
  }

  # Proxy Grafana Live WebSocket connections.
  location /grafana/api/live {
    rewrite  ^/grafana/(.*)  /$1 break;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection $connection_upgrade;
    proxy_set_header Host $http_host;
    proxy_pass http://localhost:3000/;
  }
}
```

https://user-images.githubusercontent.com/37156449/162283844-4d20183c-6350-406a-8f50-2936110a255f.mp4

**Comment by Mario-Hofstaetter** [comment]
@zuchka Thank you for your effort.

Looking at your video I noticed the URI of the Explore button is still wrong without the `/grafana/` subpath?

![grafik](https://user-images.githubusercontent.com/33002073/162391747-51cd8410-fd04-464a-b51b-c0c5aa933258.png)
Not entirely sure whats going on (redirect?) , in the browser console no fetches are visible for the /explore/ route.

However, I have since configured nginx to redirect the root url `/` to `/grafana/` so for me this is a non-issue now :-)
